### PR TITLE
[wallet/desktop] feat: unit test table display component

### PR DESCRIPTION
**Pull request by AnthonyLaw**
## What was the issue?
- missing unit test in the `TableDisplay` component.

## What's the fix?
- Added unit test  `TableDisplay` component.
- update mock data (metadata, namespace, mosaic)
- minor refactor for removed `setFilteredBy` method

### CODE REVIEW

File: __tests__/components/TableDisplay.spec.ts [hunk 1]
```diff
@@ -0,0 +1,1480 @@
...
+            runBasicOwnedAssetHexIdsTests('namespace', ['80DE90A24D6C0CC4']);
+        });
+
+        test('returns owned mosaic hex id when assert type is mosaic', () => {
+            runBasicOwnedAssetHexIdsTests('mosaic', ['75E00B2F9BDAA7EE', '49DB04D43E714646']);
+        });
+    });
+
+    // Todo: unable to mock on service class
+    // describe('getService', () => {
```
Comment (Jaguar0625) [inline __tests__/components/TableDisplay.spec.ts:288]: update or remove

File: __tests__/components/TableDisplay.spec.ts [hunk 2]
```diff
@@ -0,0 +1,1480 @@
...
+            const wrapper = getTableRowWrapper(
+                { ...params },
+                {
+                    assetType,
+                },
+            );
+
+            const vm = wrapper.vm as TableDisplayTs;
+
+            test(`returns default ${assetType} when filter expiration is hide`, () => {
```
Comment (Jaguar0625) [inline __tests__/components/TableDisplay.spec.ts:495]: in this case it shows both expired and unexpired, so test name is slightly misleading

File: __tests__/components/TableDisplay.spec.ts [hunk 3]
```diff
@@ -0,0 +1,1480 @@
...
+                const wrapper = createWrapper('pagination');
+
+                wrapper.$set(wrapper, 'currentPage', 2);
+                wrapper.$set(wrapper, 'pageSize', 2);
+
+                // Act + Assert:
+                expect(wrapper.currentPageRows).toEqual([expectedRows[2]]);
+            });
+
+            test('returns full rows when pagination type is scroll', () => {
```
Comment (Jaguar0625) [inline __tests__/components/TableDisplay.spec.ts:927]: where are you setting scroll in test?

File: __tests__/components/TableDisplay.spec.ts [hunk 4]
```diff
@@ -0,0 +1,1480 @@
...
+                    {
+                        namespaceId: new NamespaceId(mockNamespaceRowValue.name),
+                        aliasTarget: null,
+                        aliasAction: 0,
+                        mosaicId: null,
+                    },
+                );
+            });
+        });
+    });
```
Comment (Jaguar0625) [inline __tests__/components/TableDisplay.spec.ts:1088]: should you test form is not shown when there is no alias ?

File: __tests__/components/TableDisplay.spec.ts [hunk 5]
```diff
@@ -0,0 +1,1480 @@
...
+                pageSize: 10,
+            });
+        });
+
+        test('skip when asset type is namespace and current page is last page', () => {
+            runBasicSkipLoadMoreTests('namespace');
+        });
+
+        test('skip when asset type is not namespace', () => {
+            runBasicSkipLoadMoreTests('mosaic');
```
Comment (Jaguar0625) [inline __tests__/components/TableDisplay.spec.ts:1315]: should be NOT last page for this test

File: __tests__/components/TableDisplay.spec.ts [hunk 6]
```diff
@@ -0,0 +1,1480 @@
...
+
+        test('called refresh when isRefreshing status is false', async () => {
+            await runBasicRefreshingTests(false, true);
+        });
+
+        test('skip refresh when isRefreshing in progress', async () => {
+            await runBasicRefreshingTests(true, false);
+        });
+
+        // Todo: unable to catch error for coverage
```
Comment (Jaguar0625) [inline __tests__/components/TableDisplay.spec.ts:1355]: update or delete

File: __tests__/components/TableDisplay.spec.ts [hunk 7]
```diff
@@ -0,0 +1,1480 @@
...
+            runBasicInfiniteScrollDisabledTest(wrapper, true);
+        });
+
+        test('returns true when paginationType is pagination with isLoading is true', () => {
+            // Arrange:
+            const wrapper = getTableRowWrapper(
+                {
+                    isFetchingMosaics: true,
+                },
+                { paginationType: 'pagination' },
```
Comment (Jaguar0625) [inline __tests__/components/TableDisplay.spec.ts:1425]: test would be better if type is scroll + loading

File: __tests__/components/TableDisplay.spec.ts [hunk 8]
```diff
@@ -0,0 +1,1480 @@
...
+
+            // Assert:
+            expect(result).toBe(expectedResult);
+        };
+
+        test('returns true when isLoading is true and page number more than 1', () => {
+            runBasicIsFetchingMoreTests(true, 2, true);
+        });
+
+        test('returns false when isLoading is false and page number less than 1', () => {
```
Comment (Jaguar0625) [inline __tests__/components/TableDisplay.spec.ts:1472]: cosnsider isLoading == false, page number > 1

File: __mocks__/Namespaces.ts [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+        aliasTargetMosaicIdHex: undefined,
+        parentNamespaceIdHex: '',
+        startHeight: 611920,
+        endHeight: 621920,
+        depth: 0,
+        metadataList: [],
+    },
+];
+
+export const mockNamespaceRowValue = {
```
[the remark this answers is not in the document]
Comment (AnthonyLaw) [reply __mocks__/Namespaces.ts:46]: yes, I will rebase it 👍🏼

File: __mocks__/mosaics.ts [hunk 10]
```diff
@@ -290,3 +290,26 @@ export const mosaicsMock = [
         supply: 500000000,
     },
 ];
+
+export const mockMosaicRowValue = {
```
[the remark this answers is not in the document]
Comment (AnthonyLaw) [reply __mocks__/mosaics.ts:294]: yes, I will rebase it 👍🏼

File: __mocks__/mosaics.ts [hunk 11]
```diff
@@ -290,3 +290,26 @@ export const mosaicsMock = [
...
+            targetId: '80DE90A24D6C0CC4',
+            value: 'metadata',
+        },
+    ],
+    name: 'N/A',
+    restrictable: true,
+    supply: 1000,
+    supplyMutable: true,
+    transferable: false,
+};
```
[the remark this answers is not in the document]
Comment (AnthonyLaw) [reply __mocks__/mosaics.ts:315]: sure, I saw some of the files are upper class, but I think the majority are lowercase in the mocks folder.

File: __tests__/components/TableDisplay.spec.ts [hunk 12]
```diff
@@ -0,0 +1,1480 @@
...
+            prop,
+            {},
+            undefined,
+            mocks,
+        );
+    };
+
+    describe('isLoading', () => {
+        const runBasicIsLoadingStatusTests = (assetType, fetchStatus) => {
+            test(`returns false fetching ${assetType} status when asset type is namespace`, () => {
```
[the remark this answers is not in the document]
Comment (AnthonyLaw) [reply __tests__/components/TableDisplay.spec.ts:138]: hehe, typo

File: __mocks__/Metadatas.ts [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+        targetAddress: 'TBF43DIZI62PR2W6JQBJR3AI6OZLRXJYMGHLTFI',
+        scopedMetadataKey: 'E3E5E7B070607991',
+        metadataType: 2,
+        value: 'metadata',
+        targetId: '80DE90A24D6C0CC4',
+    },
+    {
+        metadataId: '6303F2DC6CF8F07F3107CEB9',
+        sourceAddress: 'TCABUWAK5WMJ26ZPERMGWBOWAJF4XPNCJOWPAAI',
+        targetAddress: 'TBF43DIZI62PR2W6JQBJR3AI6OZLRXJYMGHLTFI',
```
[the remark this answers is not in the document]
Comment (AnthonyLaw) [reply __mocks__/Metadatas.ts:14]: yeah but right now I didn't use to compared anything. maybe I will needed in future use.

File: __mocks__/Namespaces.ts [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+        ownerAddressRawPlain: 'TCABUWAK5WMJ26ZPERMGWBOWAJF4XPNCJOWPAAI',
+        aliasType: 2,
+        aliasTargetAddressRawPlain: 'TCABUWAK5WMJ26ZPERMGWBOWAJF4XPNCJOWPAAI',
+        aliasTargetMosaicIdHex: undefined,
+        parentNamespaceIdHex: '',
+        startHeight: 611920,
+        endHeight: 621920,
+        depth: 0,
+        metadataList: [],
+    },
```
[the remark this answers is not in the document]
Comment (AnthonyLaw) [reply __mocks__/Namespaces.ts:43]: yeah, i think that make sense.

File: __tests__/components/TableDisplay.spec.ts (hunk 3 above)
Comment (AnthonyLaw) [reply __tests__/components/TableDisplay.spec.ts:927]: good catch!! 👍🏼

File: __tests__/components/TableDisplay.spec.ts (hunk 4 above)
Comment (AnthonyLaw) [reply __tests__/components/TableDisplay.spec.ts:1088]: Actually no, it doesn't matter if alias exist or not, the form still allow to open.

https://github.com/symbol/desktop-wallet/blob/d28deaf0c6682c6c85c75446f67a16f0f73c2e71/src/components/TableDisplay/TableDisplayTs.ts#L463

If alias not exist, the form will let user to create `link` transaction.
If alias exist, the form will let user to create `unlink` transaction.

File: __mocks__/namespaces.ts [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+    if (aliasType === 1) {
+        aliasValue.aliasTargetMosaicIdHex = aliasTarget;
+    } else if (aliasType === 2) {
+        aliasValue.aliasTargetAddressRawPlain = aliasTarget;
+    }
+
+    return {
+        namespaceIdHex: '80DE90A24D6C0CC4',
+        name,
+        isRoot: name.split('.').length > 1 ? false : true,
```
Comment (Jaguar0625) [inline __mocks__/namespaces.ts:16]: can simplify: `name.split('.').length <= 1`

File: __tests__/components/TableDisplay.spec.ts (hunk 5 above)
Comment (Jaguar0625) [reply __tests__/components/TableDisplay.spec.ts]: i meant it should be mosaic and NOT last page (helper is always setting last page, which will skip for all types). i didn't mean to rename helper.

File: __tests__/components/TableDisplay.spec.ts [hunk 16]
```diff
@@ -0,0 +1,1436 @@
...
+
+        test('returns false when isLoading is false and page number less than 1', () => {
+            runBasicIsFetchingMoreTests(false, 0, false);
+        });
+
+        test('returns false when isLoading is true and page number less than 1', () => {
+            runBasicIsFetchingMoreTests(true, 0, false);
+        });
+
+        test('returns false when isLoading is false and page number more than 1', () => {
```
Comment (Jaguar0625) [inline __tests__/components/TableDisplay.spec.ts:1432]: more than => greater than

File: __tests__/components/TableDisplay.spec.ts (hunk 5 above)
Comment (AnthonyLaw) [reply __tests__/components/TableDisplay.spec.ts]: I see, now I got it 👍🏼

Comment (AnthonyLaw) [reply __tests__/components/TableDisplay.spec.ts]: I have rewritten a bit, it should look better now

File: __tests__/components/TableDisplay.spec.ts [hunk 17]
```diff
@@ -0,0 +1,1424 @@
...
+                    assetType,
+                },
+            );
+
+            const vm = wrapper.vm as TableDisplayTs;
+
+            // Act:
+            vm.loadMore();
+
+            if (expectedResult) {
```
Comment (Jaguar0625) [inline __tests__/components/TableDisplay.spec.ts:1240]: // Assert:

File: __tests__/components/TableDisplay.spec.ts [hunk 18]
```diff
@@ -0,0 +1,1424 @@
...
+                expect(vm.$store.dispatch).toBeCalledWith('namespace/LOAD_NAMESPACES', {
+                    pageNumber: 2,
+                    pageSize: 10,
+                });
+            }
+
+            // Assert:
+            expect(vm.$store.dispatch).not.toBeCalledWith('namespace/LOAD_NAMESPACES');
```
Comment (Jaguar0625) [inline __tests__/components/TableDisplay.spec.ts:1248]: shouldn't this be in an else ? 🤔

Comment (AnthonyLaw) [reply __tests__/components/TableDisplay.spec.ts:1248]: yeah, It should 😬

**Review by Jaguar0625 — APPROVED** [review]

**Pull request merged**